Sources/Subs-Db-mysql.php

```diff
@@ -33,44 +33,50 @@ function smf_db_initiate($db_server, $db_name, $db_user, $db_passwd, $db_prefix,
 	global $smcFunc, $mysql_set_mode;
 
 	// Map some database specific functions, only do this once.
-	if (!isset($smcFunc['db_fetch_assoc']) || $smcFunc['db_fetch_assoc'] != 'mysql_fetch_assoc')
+	if (!isset($smcFunc['db_fetch_assoc']) || $smcFunc['db_fetch_assoc'] != 'mysqli_fetch_assoc')
 		$smcFunc += array(
-			'db_query' => 'smf_db_query',
-			'db_quote' => 'smf_db_quote',
-			'db_fetch_assoc' => 'mysql_fetch_assoc',
-			'db_fetch_row' => 'mysql_fetch_row',
-			'db_free_result' => 'mysql_free_result',
-			'db_insert' => 'smf_db_insert',
-			'db_insert_id' => 'smf_db_insert_id',
-			'db_num_rows' => 'mysql_num_rows',
-			'db_data_seek' => 'mysql_data_seek',
-			'db_num_fields' => 'mysql_num_fields',
-			'db_escape_string' => 'addslashes',
-			'db_unescape_string' => 'stripslashes',
-			'db_server_info' => 'mysql_get_server_info',
-			'db_affected_rows' => 'smf_db_affected_rows',
-			'db_transaction' => 'smf_db_transaction',
-			'db_error' => 'mysql_error',
-			'db_select_db' => 'mysql_select_db',
-			'db_title' => 'MySQL',
-			'db_sybase' => false,
-			'db_case_sensitive' => false,
+			'db_query'                  => 'smf_db_query',
+			'db_quote'                  => 'smf_db_quote',
+			'db_fetch_assoc'            => 'mysqli_fetch_assoc',
+			'db_fetch_row'              => 'mysqli_fetch_row',
+			'db_free_result'            => 'mysqli_free_result',
+			'db_insert'                 => 'smf_db_insert',
+			'db_insert_id'              => 'smf_db_insert_id',
+			'db_num_rows'               => 'mysqli_num_rows',
+			'db_data_seek'              => 'mysqli_data_seek',
+			'db_num_fields'             => 'mysqli_num_fields',
+			'db_escape_string'          => 'addslashes',
+			'db_unescape_string'        => 'stripslashes',
+			'db_server_info'            => 'smf_db_get_server_info',
+			'db_affected_rows'          => 'smf_db_affected_rows',
+			'db_transaction'            => 'smf_db_transaction',
+			'db_error'                  => 'mysqli_error',
+			'db_select_db'              => 'smf_db_select',
+			'db_title'                  => 'MySQLi',
+			'db_sybase'                 => false,
+			'db_case_sensitive'         => false,
 			'db_escape_wildcard_string' => 'smf_db_escape_wildcard_string',
-			'db_is_resource' => 'is_resource',
+			'db_is_resource'            => 'smf_is_resource',
 		);
 
-	if (!empty($db_options['port']))
-		$db_server .= ':' . $db_options['port'];
-	
```

Comment:
Please keep the above code in place, as it allows connecting on a non-standard port.